aospy/data_loader.py

```diff
@@ -31,9 +38,13 @@ def rename_grid_attrs(data):
         conventions
     """
     for name_int, names_ext in internal_names.GRID_ATTRS.items():
-        data_coord_name = set(names_ext).intersection(set(data.variables))
+        dims_and_vars = set(data.variables).union(set(data.dims))
+        data_coord_name = set(names_ext).intersection(dims_and_vars)
         if data_coord_name:
             data = data.rename({data_coord_name.pop(): name_int})
+            # Force all dimensions to have coordinates.
+            if not data[name_int].coords:
+                data = data.assign_coords(**{name_int: data[name_int].values})
```

Comment:
The tests that fail are for the case where time is a scalar coordinate.  Therefore it gets caught in this if statement as well.  I think there are two options here that would fix this:
1. Change the if statement to also check if the key is a dimension in the Dataset; this will prevent a scalar time coordinate from being caught in this if statement (since it is not a dimension in the Dataset in that case):
```python
if not data[name_int].coords and name_int in data.dims:
    data = data.assign_coords(**{name_int: data[name_int].values})
```
2. Don't change the if statement (allow a scalar time to be caught), but be sure to copy over the ```attrs``` dictionary to the new coordinate (this will carry over the ```'units'``` attribute and prevent the ```KeyError``` downstream):
```python
if not data[name_int].coords:
    data_attrs = data[name_int].attrs
    data = data.assign_coords(**{name_int: data[name_int].values})
    data[name_int].attrs = data_attrs
```
Take your pick out of those solutions; perhaps the safest would actually be to combine them.  I don't see any harm copying over the ```attrs``` dictionary, and it's probably best not to modify scalar coordinates that we don't explicitly need to.